tests/ieee802154_hal/main.c

```diff
@@ -722,16 +693,170 @@ static int _caps_cmd(int argc, char **argv)
     return 0;
 }
 
+uint8_t payload[] = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam ornare lacinia mi elementum interdum ligula.";
+
+static int send(uint8_t *dst, size_t dst_len, iolist_t iol_data, size_t num, size_t time)
+{
+    uint8_t flags;
+    uint8_t mhr[IEEE802154_MAX_HDR_LEN];
+    int mhr_len;
+
+    le_uint16_t src_pan, dst_pan;
+
+    flags = IEEE802154_FCF_TYPE_DATA;// | IEEE802154_FCF_ACK_REQ;
+
+    src_pan = byteorder_btols(byteorder_htons(CONFIG_IEEE802154_DEFAULT_PANID));
+    dst_pan = byteorder_btols(byteorder_htons(CONFIG_IEEE802154_DEFAULT_PANID));
+    uint8_t src_len = IEEE802154_LONG_ADDRESS_LEN;
+    void *src = &ext_addr;
+
+    received_acks = 0;
+    received_packets = 0;
+    send_packets = 0;
+
+    for(size_t i = 0; i < num; i++) {
+        /* fill MAC header, seq should be set by device */
+        if ((mhr_len = ieee802154_set_frame_hdr(mhr, src, src_len,
+                                            dst, dst_len,
+                                            src_pan, dst_pan,
+                                            flags, seq++)) < 0) {
+            puts("txtsnd: Error preperaring frame");
+            return 1;
+        }
+
+        iolist_t iol_hdr = {
+            .iol_next = &iol_data,
+            .iol_base = mhr,
+            .iol_len = mhr_len,
+        };
+        _send(&iol_hdr);
+        xtimer_msleep(time);
+    }
+    if (ENABLE_DEBUG) {
+        puts("-------Summary of the test-------");
+        printf("Send Packets: %d\n", send_packets);
+        printf("Acknowledged Packets: %d\n", received_acks);
+        printf("Percentage: %d\n", (received_acks * 100)/num);
+        printf("Received Packets: %d\n", received_packets);
+        puts("---------------------------------");
```

Comment:
Some radios handle retransmissions automatically or don't forward the ACK frame to the upper layer.
For skipping this case, you can use the "has_xxx" functions of the Radio HAL.

In this specific case this should be enough:

```suggestion
        ieee802154_dev_t *dev = ieee802154_hal_test_get_dev(RADIO_DEFAULT_ID);
        puts("-------Summary of the test-------");
        printf("Send Packets: %d\n", send_packets);
        if (ieee802154_radio_set_frame_retrans(dev) || ieee802154_radio_has_irq_ack_timeout(dev)) {
            printf("Acknowledged Packets: %d\n", received_acks);
            printf("Percentage: %d\n", (received_acks * 100)/num);
        }
        printf("Received Packets: %d\n", received_packets);
        puts("---------------------------------");
```